app_dart/lib/src/model/ci_yaml/ci_yaml.dart

```diff
@@ -18,14 +19,34 @@ class CiYaml {
   });
 
   /// The underlying protobuf that contains the raw data from .ci.yaml.
-  final pb.SchedulerConfig config;
+  pb.SchedulerConfig config;
 
   /// The [RepositorySlug] that [config] is from.
   final RepositorySlug slug;
 
   /// The git branch currently being scheduled against.
   final String branch;
 
+  /// Load [currentConfigYaml] (and optionally [totConfigYaml] ) to [pb.SchedulerConfig] and validate the dependency graph.
+  /// where totYaml means tip of tree Yaml.
+  CiYaml.fromYaml(YamlMap? currentConfigYaml, {YamlMap? totConfigYaml})
+      : branch = '',
+        slug = RepositorySlug('flutter', 'fake'),
+        config = pb.SchedulerConfig() {
+    config.mergeFromProto3Json(currentConfigYaml);
+
+    // check for new builders and compare to tip of tree,
```

Comment:
This comment reads like it should be deleted as it was written in _validateSchedulerConfig